File: beam-scripting/sample-scripts/create-product.js
var w = 512;
var h = 512;
var p = new Product("P1", "T1", w, h);
var b1 = new VirtualBand("B1", ProductData.TYPE_FLOAT32, w, h, "sin(20*PI*(X*Y/(512*512)))");
var b2 = new Band("B2", ProductData.TYPE_FLOAT32, w, h);
b2.ensureRasterData();
p.addBand(b1);
p.addBand(b2);
var x, y, v; 
for (y = 0; y < h; y++) {
    for (x = 0; x < w; x++) {
        v = 0.1 * x * y;
        b2.setPixelFloat(x, y, v);
    }
}
visat.addProduct(p);

File: beam-scripting/sample-scripts/create-product.py
w = 512
h = 512

p = Product("P3", "T1", w, h)
b1 = VirtualBand("B1", ProductData.TYPE_FLOAT32, w, h, "sin(20*PI*(X*Y/(512*512)))")
b2 = Band("B2", ProductData.TYPE_FLOAT32, w, h)
b2.ensureRasterData()

p.addBand(b1)
p.addBand(b2)

for y in range(0, h):
    for x in range(0, w):
         v = 0.1 * x * y
         b2.setPixelFloat(x, y, v)

visat.addProduct(p)